lib/agoneum/account/session.ex

```diff
@@ -14,9 +14,27 @@ defmodule Agoneum.Account.Session do
   end
   def authenticate(_), do: {:error, :invalid_credentials}
 
+  def current_user(conn) do
+    Guardian.Plug.current_resource(conn)
+  end
+
+  def current_user_name(conn) do
+    user = current_user(conn)
+    user_display_name(user)
+  end
+
+  def logged_in?(conn), do: !!current_user(conn)
+
   @spec check_password(nil | %User{}, String.t) :: boolean()
   defp check_password(nil, _), do: Comeonin.Bcrypt.dummy_checkpw
   defp check_password(%User{} = user, password) do
     Comeonin.Bcrypt.checkpw(password, user.password_hash)
   end
+
+  defp user_display_name(user) when not is_nil(user) do
+    User.display_name(user)
+  end
+  defp user_display_name(_) do
+    "Guest"
+  end
```

Comment:
I can remove this and just make the User model responsible for it since we're just calling it anyway. I ported this over to the same spot Nermesterts had it but I get your point